src/OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.cpp

```diff
@@ -22,11 +22,12 @@ Tabulated::Tabulated(
     const Array<Instant>& anInstantArray,
     const MatrixXd& aContributionProfile,
     const Array<Shared<const CoordinatesSubset>>& aWriteCoordinatesSubsets,
-    const Shared<const Frame>& aFrameSPtr
+    const Shared<const Frame>& aFrameSPtr,
+    const InterpolationType& anInterpolationType
 )
     : Dynamics("Tabulated"),
-      contributionProfile_(aContributionProfile),
       instants_(anInstantArray),
+      contributionProfile_(aContributionProfile),
       writeCoordinatesSubsets_(aWriteCoordinatesSubsets),
       frameSPtr_(aFrameSPtr)
```

Comment:
We should check if the `aFrameSPtr` is inertial or pseudo-inertial

Reply:
hmmm what difference would that make? I plan on supporting all frames (even non-international ones) for creating the object from acceleration profiles.

Reply:
Maybe local orbital frame does not extend from `Frame` (can't remember right now), if that's the case we are good : )

Reply:
@phc1990 I think we can accept them as long as they are converted into the inertial frame at construction time (and thus not stored in a local frame). That is essentially what I have done in Maneuver [here](https://github.com/open-space-collective/open-space-toolkit-astrodynamics/blob/feat/add-maneuver-class/src/OpenSpaceToolkit/Astrodynamics/Flight/Maneuver.cpp#L56).

But good point, I can add this to TabulatedDynamics so that the interpolation and storing of the contribution profile under the hood is always in the default (GCRF) frame, good idea.

Reply:
okok I think we might be talking about different things

What I meant is that the referenceFrame given to the Maneuver Object (at construction time) or the referenceFrame given to the function toTabulatedDynamcis(...) should not be a local orbital frame.

Why?

Because the scope of the Maneuver Object is to be a historical and **reproducible** item. Having a local orbital frame will break this, as the Maneuver Objet would no longer be self contained.

So here:

```
    Maneuver(
        const Array<Instant>& anInstantArray,
        const Array<Vector3d>& anAccelerationProfile,
        const Shared<const Frame>& aFrameSPtr,
        const Array<Real>& aMassFlowRateProfile
    );
```

The `aFrameSPtr` should not be a local orbital frame.

And I don't think we should allow it at any other function either (e.g. `toTabultedDynamics(frame)`).

Just to be clear: I was not suggesting we should have `TabulatedDynamics` use GCRF by default.

Does this make sense ?

Reply:
Ah yes ok I agree. I need to double check how to prevent this cause if I recall correctly local orbital frames aren't always represented as `Shared<Frame>`.

But now that I think about it, I also think TabulatedDynamics needs to use GCRF by default, because if it uses a non-inertial frame or even a local frame we run into the same problem where it is no longer self contained, right? And since its used in the propagator, we shouldn't allow the possibility of numerically integrating in a non-inertial frame, which I think would be currently allowed.

Reply:
@phc1990 I took a brief look into this, and with the code from `LocalOrbitalFrameFactory` as it currently stands it looks like we are protected from this happening, but I took an action in JIRA to investigate how we can prevent the wrong frames from being used to specify accelerations for maneuver in my next steps

Reply:
all good!